processing/src/main/java/org/apache/druid/segment/DimensionMergerV9.java

```diff
@@ -39,6 +39,16 @@ public interface DimensionMergerV9 extends DimensionMerger
    */
   ColumnDescriptor makeColumnDescriptor();
 
+  /**
+   * Sets this merger as the "parent" of another merger for a "projection", allowing for this merger to preserve any
+   * state which might be required for the projection mergers to do their thing. This method MUST be called prior to
+   * performing any merge work
+   */
+  default void markAsParent()
```

Comment:
As to types that may exist in extensions: is this default impl ok? Let's say some type exists in an extension with its own dictionaries, and inherits this default impl. What happens then?

Reply:
I found a missing piece for this to "just work" with default implementations, this method is fine doing nothing, but `attachParent` actually needs to be modified to something like
```
  default void attachParent(DimensionMergerV9 parent, List<IndexableAdapter> projectionAdapters) throws IOException
  {
    // by default fall through to writing merged dictionary
    writeMergedValueDictionary(projectionAdapters);
  }
```
since it is called as an alternative to `writeMergedValueDictionary`, https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/segment/IndexMergerV9.java#L395, will fix in this PR